src/app/shared/models/api-workarounds.ts

```diff
@@ -0,0 +1,43 @@
+// Temporary types needed before backend fixes itself
+// Ideally, link each workaround to a JIRA after fixing which
+// it can be removed.
+
+import {
+  AccommodationOffer as AccommodationOfferFromAPI,
+  MaterialAidOffer as MaterialAidOfferFromAPI,
+  OffersAccommodationOffer as OffersAccommodationOfferFromAPI,
+  OffersMaterialAidOffer as OffersMaterialAidOfferFromAPI,
+  OffersTransportOffer as OffersTransportOfferFromAPI,
+  OffersBaseOffer as OffersBaseOfferFromAPI,
+  TransportOffer as TransportOfferFromAPI,
+} from '@app/core/api';
+
+// Remove once https://jira.sysopspolska.pl/browse/POM-299 is fixed.
+export type AccommodationOffer = Omit<AccommodationOfferFromAPI, 'type'> & {
```

Comment:
Thanks to using the same names as the "official" API, the only thing that needs changing in most files is where those types get imported from. This will make it easier to remove the workarounds when POM-299 is fixed.